rs/xnet/payload_builder/src/lib.rs

```diff
@@ -1381,7 +1403,12 @@ impl PoolRefillTask {
                         }
                     }
                 }
-            });
+            };
+            if self.block_on_async_tasks {
+                query.await;
+            } else {
+                self.runtime_handle.spawn(query);
+            }
```

Comment:
I'm not sure you need a `PoolRefillTask`. You can simply grab everything available from the other subnet and replace whatever's already in the pool (by calling `put()`). The block maker will then make sense of it and take only the bits it needs.

Reply:
> You can simply grab everything available from the other subnet and replace whatever's already in the pool

I'm not sure how this would actually work given that this function `PoolRefillTask::refill_pool` is quite complex and stuff like
```
                // XNetEndpoint only counts message bytes, allow some overhead (measuread: 350
                // bytes for certification plus base witness, 2% for large payloads).
                (slice_byte_limit.saturating_sub(350)) * 98 / 100,
```
looks brittle when I do something ad-hoc.